c10/util/AbortHandler.h

```diff
@@ -73,7 +73,7 @@ C10_ALWAYS_INLINE void set_terminate_handler() {
   use_custom_terminate = true;
 #endif // _WIN32
   auto result = c10::utils::check_env("TORCH_CUSTOM_TERMINATE");
-  if (result != std::nullopt) {
+  if (result.has_value()) {
```

Comment:
Nit (`std::optional` has [`operator bool`](https://en.cppreference.com/w/cpp/utility/optional/operator_bool)) aren't it?
```suggestion
  if (result) {
```

Reply:
Clang-tidy check bugprone-unchecked-optional-access likes explicit has_value()